metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1175,104 +1175,43 @@ def get_tables_for_dataset(
-                # For sharded tables we only process the latest shard and ignore the rest
-                # to find the latest shard we iterate over the list of tables and store the maximum shard id
-                # We only has one special case where the table name is a date `20220110`
-                # in this case we merge all these tables under dataset name as table name.
-                # For example some_dataset.20220110 will be turned to some_dataset.some_dataset
-                # It seems like there are some bigquery user who uses this non-standard way of sharding the tables.
-                if shard:
-                    if not sharded_tables.get(table_name):
-                        # When table is only a shard we use dataset_name as table_name
-                        sharded_tables[table_name] = table
-                        continue
-
-                    stored_table_identifier = BigqueryTableIdentifier(
-                        project_id=project_id,
-                        dataset=dataset_name,
-                        table=sharded_tables[table_name].table_id,
-                    )
-                    _, stored_shard = BigqueryTableIdentifier.get_table_and_shard(
-                        stored_table_identifier.raw_table_name()
-                    )
-                    # When table is none, we use dataset_name as table_name
-                    assert stored_shard
-                    if stored_shard < shard:
-                        sharded_tables[table_name] = table
-                    continue
-                elif str(table_identifier).startswith(
-                    self.config.temp_table_dataset_prefix
-                ):
-                    logger.debug(f"Dropping temporary table {table_identifier.table}")
-                    self.report.report_dropped(table_identifier.raw_table_name())
-                    continue
-                else:
-                    if (
-                        table.time_partitioning
-                        or "range_partitioning" in table._properties
-                    ):
-                        partitioned_table_count_in_this_batch += 1
-
-                    table_items[table.table_id] = table
-
-                if (
-                    len(table_items) % self.config.number_of_datasets_process_in_batch
-                    == 0
-                ) or (
-                    partitioned_table_count_in_this_batch
-                    == self.config.number_of_partitioned_datasets_process_in_batch
-                ):
+            # We get the list of tables in the dataset to get core table properties and to be able to process the tables in batches
+            # We collect only the latest shards from sharded tables (tables with _YYYYMMDD suffix) and ignore temporary tables
+            table_items = self.get_core_table_details(conn, dataset_name, project_id)
+
+            items_to_get: Dict[str, TableListItem] = {}
+            for table_item in table_items.keys():
```

Comment:
So ultimately we're querying `INFORMATION_SCHEMA.TABLES` (perhaps joined with `INFORMATION_SCHEMA.PARTITIONS`) with some filter `and t.table_name in ...`. Is there any way to order the tables so we're fetching data most efficiently, or not really?

I guess I'm a bit confused how accessing the PARTITIONS table causes us to query more tables, unless each partition counts as its own table (or maybe a % of a table? lol)

Reply:
On Partitions table has this limitation the other information schemas don't and that's why only this table is affected. We don't query more tables here than in any other information schema table.

Reply:
Is this documented on https://cloud.google.com/bigquery/docs/information-schema-partitions? Or just something you have to find out lol